data/teca_calendar_util.h

```diff
@@ -167,6 +167,8 @@ class TECA_EXPORT interval_iterator
     virtual int get_next_interval(time_point &first_step,
         time_point &last_step) = 0;
 
+    virtual int set_number_of_steps(long n) {}
```

Comment:
this is specific to the n_steps iterator and therefor should not be a virtual in the base class because the base class is for api common to all the iterators.